README.org

```diff
@@ -6,40 +6,38 @@ inspired by Emacs.
 
 The main types are:
 
-- A =key= structure has a hardware =code=, a symbolic =value=, an ordered set of
-  modifiers (the order is to avoid inequality on permutation) and possibly more
-  in the future.
+- A =key= structure that holds:
+  - a numeric =code= :: a hardware keycode;
+  - a string =value= :: a character (except for whitespaces);
+  - an ordered set of modifiers (to ensure equality on permutation).
 
-  Whitespaces are the only reserved characters.  Anything else can be stored in
-  the =key= value.
+- A =keyspec= is a string representation of a =key= (e.g. "control-a").  A
+  =keyspecs= is the string concatenation of two or more =keyspec=s
+  (e.g. "control-a control-b control-c").
 
-- A =keyspec= is a string representation of a =key=.  A =keyspec= is a
-  space-separated string representation of a sequence of =key=s.
-
-- A =keymap= contains a hash-table where the keys are a =keyspec= and the values
-  are either a keymap or anything else.  As a special case, a =nil= value
-  unbinds the key.
+- A =keymap= contains a hash-table that maps =keyspecs= to either keymaps,
+  symbols or nil (to unbind the =keyspecs=).
```

Comment:
Note that there can be other values that one can put there:
- `nyxt:command`s,
- `functions`s.
- And anything else, actually. `nkeymaps` is quite lenient about that.

So maybe we shouldn't mention it here and in the docstrings...